lib/src/chip/filter-button/filter-button.component.ts

```diff
@@ -0,0 +1,22 @@
+import { Component, HostBinding, Input, OnChanges } from '@angular/core';
+
+@Component({
+    selector: 'hc-filter-button',
+    template: `<img src="./assets/FilterIcon.svg" class="filter-img">{{buttonLabel}}`,
+    styleUrls: ['../chip.component.scss']
+})
+export class FilterButtonComponent implements OnChanges  {
+    @HostBinding('class') hostClass = 'filter-button';
+
+    @Input() labelOff: string = 'Filters off';
+    @Input() labelOn: string = 'Filters on';
+    @Input() buttonState: boolean = false;
+    buttonLabel: string = this.labelOff;
+
+    constructor() {
+    }
+
+    ngOnChanges() {
+        this.buttonLabel = (this.buttonState) ? this.labelOn : this.labelOff;
```

Comment:
It's easier to handle this in the setter function for the buttonState so move this logic there